policy/modules/system/systemd.if

```diff
@@ -58,28 +69,603 @@ template(`systemd_role_template',`
 	# This domain is per-role because of the below transitions.
 	# See the systemd --user section of systemd.te for the
 	# remainder of the rules.
-	allow $1_systemd_t $3:process { setsched rlimitinh };
+	allow $1_systemd_t $3:process { setsched rlimitinh signal_perms };
+	allow $1_systemd_t self:netlink_kobject_uevent_socket create_socket_perms;
 	corecmd_shell_domtrans($1_systemd_t, $3)
 	corecmd_bin_domtrans($1_systemd_t, $3)
-	allow $1_systemd_t self:process signal;
+	allow $1_systemd_t self:process { getsched signal };
 
 	files_search_home($1_systemd_t)
 
+	init_link_key_template($1_systemd_t)
+
+	miscfiles_watch_localization($1_systemd_t)
+
+	mount_read_runtime_files($1_systemd_t)
+	mount_watch_runtime_files($1_systemd_t)
+	mount_watch_reads_runtime_files($1_systemd_t)
+
+	allow $3 $1_systemd_t:process signal;
 	# Allow using file descriptors for user environment generators
 	allow $3 $1_systemd_t:fd use;
 	allow $3 $1_systemd_t:fifo_file rw_inherited_fifo_file_perms;
+	allow $3 $1_systemd_t:unix_stream_socket { connectto rw_socket_perms };
+
+	allow $1_systemd_t self:unix_stream_socket { connectto rw_socket_perms };
```

Comment:
this should just be `allow $1_systemd_t self:unix_stream_socket create_stream_socket_perms;` no need for connectto AFAIK

Reply:
I get this repeated denial without it:
```
time->Fri Mar 12 13:05:31 2021
type=PROCTITLE msg=audit(1615572331.762:719): proctitle="(systemd)"
type=PATH msg=audit(1615572331.762:719): item=0 name="/run/user/1000/bus" inode=15 dev=00:26 mode=0140666 ouid=1000 ogid=1000 rdev=00:00 obj=staff_u:object_r:session_dbusd_runtime_t:s0 nametype=NORMAL cap_fp=0 cap_fi=0 cap_fe=0 cap_fver=0 cap_frootid=0
type=CWD msg=audit(1615572331.762:719): cwd="/"
type=SOCKADDR msg=audit(1615572331.762:719): saddr=01002F72756E2F757365722F313030302F62757300
type=SYSCALL msg=audit(1615572331.762:719): arch=c000003e syscall=42 success=no exit=-13 a0=4 a1=38350630 a2=15 a3=79fa50634794 items=1 ppid=1 pid=1897 auid=1000 uid=1000 gid=1000 euid=1000 suid=1000 fsuid=1000 egid=1000 sgid=1000 fsgid=1000 tty=(none) ses=2 comm="systemd" exe="/lib/systemd/systemd" subj=staff_u:staff_r:staff_systemd_t:s0-s0:c0.c1023 key=(null)
type=AVC msg=audit(1615572331.762:719): avc:  denied  { connectto } for  pid=1897 comm="systemd" path="/run/user/1000/bus" scontext=staff_u:staff_r:staff_systemd_t:s0-s0:c0.c1023 tcontext=staff_u:staff_r:staff_dbusd_t:s0-s0:c0.c1023 tclass=unix_stream_socket permissive=0
```
But it may just be cosmetic? I will see what (if anything) breaks without this access.

Reply:
I think the rework to dbus accesses solved this. Nonetheless I can't reproduce the `connectto` access.